mcp/src/test/java/io/modelcontextprotocol/spec/McpSchemaTests.java

```diff
@@ -449,6 +450,106 @@ void testGetPromptResult() throws Exception {
 
 	// Tool Tests
 
+	@Test
+	void testJsonSchema() throws Exception {
+		String schemaJson = """
+				{
+					"type": "object",
+					"properties": {
+						"name": {
+							"type": "string"
+						},
+						"address": {
+							"$ref": "#/$defs/Address"
+						}
+					},
+					"required": ["name"],
+					"$defs": {
+						"Address": {
+							"type": "object",
+							"properties": {
+								"street": {"type": "string"},
+								"city": {"type": "string"}
+							},
+							"required": ["street", "city"]
+						}
+					}
+				}
+				""";
+
+		McpSchema.JsonSchema schema = mapper.readValue(schemaJson, McpSchema.JsonSchema.class);
+
+		assertThat(schema.type()).isEqualTo("object");
+		assertThat(schema.properties()).containsKeys("name", "address");
+		assertThat(schema.required()).containsExactly("name");
+		assertThat(schema.defs()).isNotNull();
+		assertThat(schema.defs()).containsKey("Address");
```

Comment:
Personally I think its easier to serialize object -> String, deserialize string -> object, then, serialize again, Then compare 1st and third string :) Saves a lot of typing and doesnt run the risk of someone else forgetting to add assets here when new properties are added!